libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -266,6 +267,51 @@ std::string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const*
 	});
 }
 
+std::string YulUtilFunctions::requireWithError(FunctionCall const& errorConstructorCall)
+{
+	std::string functionName = "require_helper_error"; // TODO: this is generated as is in IR, instead of having the hash appended to the function name
```

Comment:
Well, it doesn't just magically appear - you'll need to add it to the name :-).
See e.g.
```
	std::string functionName =
		std::string(_assert ? "assert_helper" : "require_helper") +
		(_messageType ? ("_" + _messageType->identifier()) : "");
```
in the existing ``requireOrAssertFunction`` function.

We'll need to be very careful with this here. The function name needs to be unique wrt the error type, but also the entire list of argument types to the error constructor call. That's also a test case: have two require statements to the *same* error (e.g. with a ``string`` argument), but once pass a ``string memory`` and the other time a string literal to it - that has to result in two distinct ``require_helper`` functions (since for encoding string literals we don't need an additional arguments, but in the ``string memory`` case we need to pass the memory location down to pass on to abi encoding)

Reply:
Have a look at ``ABIFunctions::tupleEncoder`` and the ``functionName`` in there for example.
What you'll need to have in the name is:

A unique identifier for the error (taking into account overloads, so something that uniquely identifies the precise error including its arguments) plus the types of the expressions in the arguments to the ``errorConstructorCall``.